src/io/procfs.rs

```diff
@@ -75,7 +75,16 @@ fn check_proc_entry_with_stat(
     // Check the ownership of the directory. We can't do that for the toplevel
     // "/proc" though, because in e.g. a user namespace scenario, root outside
     // the container may be mapped to another uid like `nobody`.
-    if !matches!(kind, Kind::Proc) && (entry_stat.st_uid, entry_stat.st_gid) != (uid, gid) {
+    //
+    // In addition to allowing the expected uid:gid, also allow root:root, as
```

Comment:
I'm still a bit skeptical of the value of these checks versus just verifying the filesystem type is `procfs`, FWIW.

Reply:
It's admittedly murky. Linux doesn't really make any guarantees about what procfs *won't* do in future versions. The idea here was to check everything we practically can to hopefully catch anything unusual.

Thinking about this more, I wouldn't be surprised if some combination of user namespaces and `PR_SET_DUMPABLE` could cause even the new check here to spuriously fail. So maybe we should remove the uid check altogether.

Reply:
I've now submitted https://github.com/bytecodealliance/rustix/pull/450 removing the uid/gid checks entirely.